react/week3/todo/src/components/TodoItem.js

```diff
@@ -0,0 +1,50 @@
+import React, { useState } from "react";
+//typecheck
+import PropTypes from "prop-types";
+
+const TodoItem = (props) => {
+  const { todoId, todoDescription, todoDeadline } = props;
+  const [check, setCheck] = useState(false);
+  const [updatedTodo, setUpdatedTodo] = useState(todoDescription);
+  const [input, setInput] = useState(false);
+  const isDone = () => setCheck(!check);
+  const [edit, setEdit] = useState(true);
+
+  const updateInputState = () => {
+    setInput(!input);
+    setEdit(!edit);
+  };
+
+  return (
+    <div>
+      <li key={todoId}>
+        {input ? (
+          <input
+            type="text"
+            checked={check}
+            onChange={(e) => setUpdatedTodo(e.target.value)}
+          />
+        ) : (
+          updatedTodo
+        )}
+        | {todoDeadline} <input type="checkbox" onClick={isDone} />{" "}
```

Comment:
Here you could simply refer to setCheck --> `onClick={() => setCheck(true)}`

However this only sets check to `true`. I think you want to be able to uncheck it as well when the user clicks again. Then you could say when check is true, set it to false and vice versa. So always set it to the opposite of the current state. Do you have an idea how to do that? :slightly_smiling_face:

Reply:
It actually worked with unchecking the box when I refered to {todoDeadline} <input type="checkbox" onClick={() => setCheck(true)} /> 

So, I'm not sure what needs to be done now :)

Reply:
Oh you're right, sorry it's my bad, my eyes somehow missed the exclamation mark. :sweat_smile:  

You could dynamically add a strike-through over the completed todo - like the one they have on the gif in the task description - but only if you're up for it. :slightly_smiling_face: